src/target/drivers/filesystems/syscalls.c

```diff
@@ -97,10 +95,10 @@ void FS_Unmount()
 
 int FS_OpenDir(const char *path)
 {
-    FATFS *ptr = &fat[0];
+    FATFS *ptr = &drive[0].fat;
```

Comment:
maybe this can use a MACRO to make the logic explicit like if (IN_MEDIA_DRIVER)? or
define a function:
 drive_num = get_drive_index_from_filename();
ptr = drive[drive_num].fat;

I don't really like this ifdef. it is not that special.